uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js

```diff
@@ -261,7 +262,7 @@ define([
           } else if (result.files.length && !!result.files[0].values[0].result) {
             DocumentService.save({
               id: result.document.id,
-              status_id: '3' // 3 => 'approved'
+              status_id: ($scope.isRole('admin')) ? '3' : '1' // 3 => 'approved', 1 => 'awaiting upload'
```

Comment:
> Why is the status id hardcoded?

The status id `2` which  corresponds to `Awaiting Approval` is hardcoded, as this is constant and never changes.

>So, for staff and manager the status is always going to be `Awaiting Approval`?

Not really, the status will be `Awaiting Approval` if the document is updated with attachments attached, otherwise the status will be `Awaiting Upload`